File: SunkiojiDalis/src/Items/Equipables/Weapons/WeaponRarities/CommonWeapon.ts
import { AbstractWeapon } from '../AbstractWeapon';

export class CommonWeapon extends AbstractWeapon {
  public Id: number;
  public Sprite: string;
  public Name: string;
  public Weight: number;
  public Quantity: number;
  public X: number;
  public Y: number;
  public BelongsTo = -1;
  public Damage: number;

  constructor(guid: string, itemData: any) {
    super(guid, itemData);
  }
}

File: SunkiojiDalis/Models/Items/Equipables/Weapons/WeaponRarities/CommonWeapon.cs
using System.Collections.Generic;
using SignalRWebPack.Engine;

namespace SignalRWebPack{
  public class CommonWeapon: AbstractWeapon {
    public CommonWeapon(int id, string sprite, string name, int weight, int quantity, int x, int y, int belongsTo, int damage) {
      this.Id = id;
      this.Sprite = sprite;
      this.Name = name;
      this.Weight = weight;
      this.Quantity = quantity;
      this.X = x;
      this.Y = y;
      this.BelongsTo = -1;
      this.AreaId = AreaId;
      this.Damage = damage;
    }
    public override Dictionary<string, string> OnClientSideCreation() {
      Dictionary<string, string> commonWeaponData = base.OnClientSideCreation();
      commonWeaponData["damage"] = this.Damage.ToString();
      commonWeaponData["objectType"] = nameof(ServerObjectType.CommonWeapon);
      return commonWeaponData;
    }
  }
}